presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -1342,7 +1342,72 @@ public Node visitNullLiteral(SqlBaseParser.NullLiteralContext context)
     @Override
     public Node visitStringLiteral(SqlBaseParser.StringLiteralContext context)
     {
-        return new StringLiteral(getLocation(context), unquote(context.STRING().getText()));
+        return new StringLiteral(getLocation(context), handleStringExpression(context.stringExpression()));
+    }
+
+    private static String handleStringExpression(SqlBaseParser.StringExpressionContext context)
```

Comment:
Name this with something more descriptive like "decode" or "unquote" rather than "handle" (also, the name "handle" usually implies it has side effects)